File: admin/src/main/java/top/zzf4/blog/entity/model/IpStat.java
package top.zzf4.blog.entity.model;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@TableName("ip_stats")
@NoArgsConstructor
@AllArgsConstructor
public class IpStat {
    @TableId(value = "id", type = IdType.AUTO)
    private Integer id;

    @TableField("ip_address")
    private String ipAddress;

    private String city;

    @TableField(value = "access_count")
    private Integer accessCount;

    @TableField(value = "banned_count")
    private Integer bannedCount;

    @TableField(value = "ban_status")
    private Integer banStatus;
}

File: blog-backend/internal/model/ip_stat.go
package model

// IPStat 对应于数据库中的 `ip_stats` 表
type IPStat struct {
	BaseModel   `gorm:"embedded"`
	IPAddress   string `gorm:"column:ip_address;uniqueIndex" json:"ipAddress"`   // IP 地址，唯一
	City        string `gorm:"column:city" json:"city"`                          // 城市
	AccessCount int    `gorm:"column:access_count;default:0" json:"accessCount"` // 访问次数
	BannedCount int    `gorm:"column:banned_count;default:0" json:"bannedCount"` // 封禁次数
	BanStatus   bool   `gorm:"column:ban_status;default:false" json:"banStatus"` // 封禁状态
}

// TableName 指定 GORM 使用的表名
func (IPStat) TableName() string {
	return "ip_stats"
}
